app/jobs/assignment_repo/create_github_repository_job.rb

```diff
@@ -16,14 +16,18 @@ class CreateGitHubRepositoryJob < ApplicationJob
     #
     # rubocop:disable MethodLength
     # rubocop:disable AbcSize
+    # rubocop:disable CyclomaticComplexity
+    # rubocop:disable PerceivedComplexity
     def perform(assignment, user)
       start = Time.zone.now
 
-      return unless assignment.invitation&.waiting? \
-        || assignment.invitation&.errored_creating_repo? \
-        || assignment.invitation.status.nil?
+      invite_status = assignment.invitation&.status(user)
```

Comment:
Looking at the code below, we're returning if this is nil, but we're still guarding against nil with `&.`. What if right below this line, we added `return unless invite_status`, and removed all the nil guards?

Reply:
I instead validate the presence of an `AssignmentInvitation` on an `Assignment`. `Assignment`s are always created with an invitation:
https://github.com/education/classroom/blob/ebd4d33fc9eaeaba052f8b5716cc1acf92c64fc4/app/controllers/assignments_controller.rb#L14-L17

By validating the presence of an invitation we won't need to guard on the invitation anymore.